flutter/test/navigation/time_to_display_tracker_test.dart

```diff
@@ -0,0 +1,173 @@
+                element.context.operation ==
+                SentrySpanOperations.uiTimeToInitialDisplay)
+            .first;
+
+        expect(transaction, isNotNull);
+        expect(transaction.context.operation, SentrySpanOperations.uiLoad);
+        expect(transaction.startTimestamp, ttidSpan.startTimestamp);
+      });
+
+      test('startMeasurement finishes ttid span', () async {
+        SentryFlutter.native = TestMockSentryNative();
+        final sut = fixture.getSut();
+        final endTimestamp =
+            fixture.startTimestamp.add(const Duration(milliseconds: 10));
+
+        await sut.trackAppStartTTD(fixture.getTransaction(name: '/'),
+            startTimestamp: fixture.startTimestamp, endTimestamp: endTimestamp);
+
+        await Future.delayed(const Duration(milliseconds: 100));
+
+        final transaction = fixture.hub.getSpan() as SentryTracer;
+
+        final spans = transaction.children;
+        expect(transaction.children, hasLength(1));
+        expect(spans[0].context.operation,
+            SentrySpanOperations.uiTimeToInitialDisplay);
+        expect(spans[0].finished, isTrue);
+      });
+    });
+
+    group('in regular routes', () {
+      test('matches startTimestamp of transaction', () async {
+        final sut = fixture.getSut();
+
+        await sut.trackRegularRouteTTD(fixture.getTransaction(),
+            startTimestamp: fixture.startTimestamp);
+
+        final transaction = fixture.hub.getSpan() as SentryTracer;
+
+        final spans = transaction.children;
+        final ttidSpan = spans
+            .where((element) =>
+                element.context.operation ==
+                SentrySpanOperations.uiTimeToInitialDisplay)
+            .first;
+
+        expect(transaction, isNotNull);
+        expect(transaction.context.operation, SentrySpanOperations.uiLoad);
+        expect(transaction.startTimestamp, ttidSpan.startTimestamp);
+      });
+
+      group('with approximation strategy', () {
+        test('startMeasurement finishes ttid span', () async {
+          final sut = fixture.getSut();
+
+          await sut.trackRegularRouteTTD(fixture.getTransaction(),
+              startTimestamp: fixture.startTimestamp);
+
+          final transaction = fixture.hub.getSpan() as SentryTracer;
+          await Future.delayed(const Duration(milliseconds: 2000));
+
+          final spans = transaction.children;
+          expect(transaction.children, hasLength(1));
+          expect(spans[0].context.operation,
+              SentrySpanOperations.uiTimeToInitialDisplay);
+          expect(spans[0].finished, isTrue);
+        });
+      });
+
+      group('with manual strategy', () {
+        test('finishes ttid span after reporting with manual api', () async {
+          final sut = fixture.getSut();
+
+          Future.delayed(const Duration(milliseconds: 100), () {
+            fixture.ttidTracker.markAsManual();
+            fixture.ttidTracker.completeTracking();
+          });
```

Comment:
```suggestion
          fixture.ttidTracker.markAsManual();
          Future.delayed(const Duration(milliseconds: 1), () {
            fixture.ttidTracker.completeTracking();
          });
```
all other delays in all tests of this class seems useless, as they pass even without delays

Reply:
oh yeah my bad, the delays are not needed anymore since await already waits for everything to finish, wasn't the case before